### PR TITLE
 Generalize jinja filters to accept lists of data objects

### PR DESCRIPTION
**Pull request by mzganec**

### CODE REVIEW

File: resolwe/flow/expression_engines/jinja/filters.py [hunk 1]
```diff
@@ -23,16 +23,22 @@ def _get_data_attr(data, attr):
 
 def name(data):
     """Return `name` of `Data`."""
+    if isinstance(data, (list, tuple)):
```
Comment (dblenkus) [inline resolwe/flow/expression_engines/jinja/filters.py]: Move the whole logic to a helper function, so you don't need to repeat it in each function.

Comment (kostko) [reply resolwe/flow/expression_engines/jinja/filters.py]: I agree with @dblenkus as it is repeating quite a lot here. Perhaps have something like `apply_filter_list(value, lambda item: _get_data_attr(item, 'name'))` and then you can handle both list and non-list cases in a generic fashion.

File: resolwe/flow/expression_engines/jinja/filters.py [hunk 2]
```diff
@@ -91,6 +91,23 @@ def descriptor(obj, path=''):
     return resp
 
 
+def apply_filter_list(func, objs):
```
Comment (dblenkus) [inline resolwe/flow/expression_engines/jinja/filters.py]: Define this function before it is used.

**Comment by mstajdohar** [comment]
Looks good, thanks.

**Pull request merged**